pandas/core/arrays/datetimelike.py

```diff
@@ -823,25 +826,41 @@ def _validate_searchsorted_value(self, value):
         return value
 
     def _validate_setitem_value(self, value):
-        if lib.is_scalar(value) and not isna(value):
-            value = com.maybe_box_datetimelike(value)
 
         if is_list_like(value):
-            value = type(self)._from_sequence(value, dtype=self.dtype)
-            self._check_compatible_with(value, setitem=True)
-            value = value.asi8
-        elif isinstance(value, self._scalar_type):
-            self._check_compatible_with(value, setitem=True)
-            value = self._unbox_scalar(value)
+            value = array(value)
+            if is_string_dtype(value.dtype):
+                # We got a StringArray
```

Comment:
is this assumption important or is value pre-validated

```
>>> value = [1, 2, object()]
>>> value = pd.array(value)
>>> value
<PandasArray>
[1, 2, <object object at 0x0000015F4B631DA0>]
Length: 3, dtype: object
>>>
>>> pd.core.dtypes.common.is_string_dtype(arr.dtype)
True
>>>
```

might be worth adding type annotations to _validate_setitem_value

Reply:
yikes, that looks like a problem with is_string_dtype

Reply:
@jbrockmendel addressing here? or followup?

Reply:
follow-up

Reply:
That's not a "problem" with `is_string_dtype`, because that function simply checks for object dtype. That's how it has been for a long time, and not sure if it is easy to solve, as actually checking for strings would mean to infer the dtypes from the values each time. We have https://github.com/pandas-dev/pandas/issues/15585 about this.

(it's a problem in practice of course, since it makes this function rather useless. But the solution is probably improving the new "string" dtype so this can become the default string dtype in the future)

Reply:
right actually I would say this is a bug in `is_string_dtype` which is actually just checking for `object` dtype and not an inferred string itself.

Reply:
you can't infer string values from a dtype object, though ..

Reply:
sure you can, this is *excatly* what `infer_from_dtype` does; we just don't do it *inside* this function because it can be non-cheap

Reply:
updated to punt on is_string_dtype by instead checking `is_dtype_equal(value.dtype, "string")`